### ISSUE TITLE
about beta hyperparameter

### ISSUE THREAD

**Issue by PureMmm** [sniklaus/softmax-splatting#31]

**Comment by PureMmm** [comment]
Hi, Snikaus. Please forgive my poor English.
I saw the hyperparameter beta was set to -20 which is mentioned in other issue too , and I also saw the beta parameter in other's reproduction code. However, I haven't found it in the paper. The only beta parameter i found is in the supplementary material , which is used for proving the translation invariance, and the beta was not used in the formula (cancelled out). So I wonder where the beta parameter refers to specifically? As shown in the following code, is this beta1 dispensable?
![image](https://user-images.githubusercontent.com/79971097/110200339-61733100-7e98-11eb-928d-3f39b13a1af4.png)

I only find the alpha parameter, but it doesn't seem to be the same thing...
![image](https://user-images.githubusercontent.com/79971097/110200235-dd20ae00-7e97-11eb-9eac-0e8d562b464d.png)

**Comment by sniklaus** [comment]
Thank you for you message! My apologies, the comment in the code should have said "alpha" and not "beta". I updated it accordingly. I am not sure where the code that you referenced originates form, but it looks sound to me. The equation you referenced (equation 14 in our paper) scales the importance metric since the scale influences how the blending is done (roughly, a small alpha yields averaging while a large alpha yields z-buffering). This seems to be exactly what we are seeing in the paper, an importance metric scaled by a (learned) scalar value. It should have been called "alpha" though and not "beta", but that is probably due to my mistake in the comment.

**Comment by PureMmm** [comment]
Thanks for your reply! I would like to know if the paramScale and beta1 are the same parameter, both refer to the alpha in the text, but they seem to be used repeatedly in the code. I need to delete one of them. Is this the meaning?
![image](https://user-images.githubusercontent.com/79971097/110284903-6ca39980-801d-11eb-9b70-3144185f4ad4.png)
![image](https://user-images.githubusercontent.com/79971097/110285344-171bbc80-801e-11eb-9b05-e34c4114b2a3.png)
![image](https://user-images.githubusercontent.com/79971097/110284953-7c22e280-801d-11eb-80c6-6ba128f3581a.png)

**Comment by sniklaus** [comment]
My apologies, but you are referencing other people's code so I am unable to answer your question. Multiplying two parameters seems redundant though, but there is no way for me to tell since the code you referenced isn't mine.

**Comment by PureMmm** [comment]
These two parts of code are provided by you, indeed. The first picture is the code in run.py, and I just changed -20 into a parameter beta that can be learned according to your comments. Then the code in the second picture is that you replied to others in other issue. [Issues4](https://github.com/sniklaus/softmax-splatting/issues/4)
![image](https://user-images.githubusercontent.com/79971097/110402977-49c5c380-80b7-11eb-939b-a86e9db034e8.png)
![image](https://user-images.githubusercontent.com/79971097/110402992-521dfe80-80b7-11eb-9ca3-787f97199f27.png)



Your work is very impressive, I would like to ask another question, you have two parts softmax operation (1 to 0 and 0 to 1) in the network structure of the paper, my understanding is that only two parameters alpha need to be learned, do I need to set two parameters beta1 and beta2 to be learned respectively?Again they use one parameter. How did you implement it?

**Comment by sniklaus** [comment]
In the `run.py` we compute the L1 in line 52 and then multiply with a fixed alpha in line 58 as follows.

https://github.com/sniklaus/softmax-splatting/blob/a973e3e94a1f9578c7db0af9cb1863f658ec07e1/run.py#L52

https://github.com/sniklaus/softmax-splatting/blob/a973e3e94a1f9578c7db0af9cb1863f658ec07e1/run.py#L58

In issue #4, the learned alpha is multiplied in the same line as the L1 computation. Hope that explanation helps.

Regarding your second question, if you apply softsplat twice (once from frame 1 towards frame 0, and once from frame 0 towards frame 1) and you compute the (importance) metric in the same way then there is no need to have two alphas, just a single one suffices. Regarding your question about beta1 and beta2, which beta are you referring to? I hope that my initial response cleared up that "beta" in the comment in `run.py` should have been "alpha" instead.

**Comment by PureMmm** [comment]
Thank for your reply。I'm sorry I didn't make it clear.By beta I mean the alpha parameter. Beta1 and Beta2 correspond to the alpha parameter of SoftSplat twice, which is now unnecessary. It helps me a lot, Thank you again.
